### DISCUSSION TITLE
Add more judgement windows to the osu ruleset

### DISCUSSION THREAD

**Discussion by Rekunan** [ppy/osu#15899] [Ideas]
I feel that the 100s window can be too broad, being able to get tapping from proper but slightly inaccurate to borderline improper. This is made more apparent on high OD values, especially OD11+.

Just adding 2 more should be fine, since most rhythm games have at least 5, and probably have it be similar/identical to mania's judgements.

I don't have enough experience/knowledge with the other rulesets to comment on them, but I wouldn't be surprised if the same applies (namely taiko).

**Comment by respektive** [comment]
I think making changes like this to the ruleset would be pretty weird after so many years, however i think adding a mod that will make the judgements harder (or add more if that's possible for mods?) would work best for this.

**Reply by integerrr** [reply]
If we're gonna go there why not delete all judgement windows and use the offset straight for scoring purposes? Although that's possibly a lot more data required for score submission, it sounds fun tho

**Comment by abraker95** [comment]
Why not have a continuous distribution?

**Reply by bdach** [reply]
i'm aware this is a vsrg thing but it does make understanding a lot more difficult for new players, and also kind of complicates a lot of the existing structure (say, what do you show on the website on beatmap pages in the score table instead of 300/100/50 count?). also i'm told even vsrg players are not unanimous on this

**Reply by abraker95** [reply]
It shouldn't complicate anything. You can still have the 300/100/50 count as visual feedback for the player based on the hit's offset, but process the score with respect to the continuous distribution in the background. Acc/score is the only thing it should effect.

![](https://i.imgur.com/5PW8pX4.png)

> also i'm told even vsrg players are not unanimous on this

The not unanimous part is the difficulty in getting a 100% since it's not humanely possible getting 0 ms for all notes. You have to clip and renormalize the distribution like this to fix that

![](https://i.imgur.com/hPcickZ.png)

The main advantage in doing it continuously instead of discretely is mathematical accuracy. You can pretty much recover the original distribution using just mean and deviation. Meanwhile the recovered distribution is much less accurate to the original one if using discrete scoring, especially if the score weights don't perfectly follow a normal distribution. Being able to recover the distribution more accurately would allow pp calc to be more accurate as well.
